src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+						final ArrayList<String> frontierRules = new ArrayList<String>();
+						final NodeList productionFrontierChildNodes = productionRule.getChildNodes();
+						for (int pr_i = 0; pr_i < productionFrontierChildNodes.getLength(); ++pr_i) {
+							final Node productionFrontierChildNode = productionFrontierChildNodes.item(pr_i);
+							if (productionFrontierChildNode.getNodeName().equals("frontierRules")) {
+								frontierRules.add(productionFrontierChildNode.getAttributes().getNamedItem("name").getNodeValue().substring(3));
+							}
+						}
+						s_productionFrontiers.put(playerName, frontierRules);
+					}
+				}
+				stepToGo = Math.max(stepToGo, (s_productionFrontiers.isEmpty() ? 5 : 9));
+			} else if (childNodeName.equals("gamePlay")) {
+				final NodeList gamePlayChildNodes = childNode.getChildNodes();
+				for (int p_i = 0; p_i < gamePlayChildNodes.getLength(); ++p_i) {
+					final Node gamePlayChildNode = gamePlayChildNodes.item(p_i);
+					if (gamePlayChildNode.getNodeName().equals("delegate")) {
+						final HashMap<String, String> attrDelegate = getAttributesMap(gamePlayChildNode.getAttributes());
+						final ArrayList<String> newValues = new ArrayList<String>();
+						newValues.add(attrDelegate.get("javaClass").replace(TRIPLEA_JAVA_CLASS_DELEGATE_PATH, ""));
+						newValues.add(attrDelegate.get("display"));
+						s_gamePlaySequence.put(attrDelegate.get("name"), newValues);
+					} else if (gamePlayChildNode.getNodeName().equals("sequence")) {
+						final NodeList sequenceChildNodes = gamePlayChildNode.getChildNodes();
+						for (int seq_i = 0; seq_i < sequenceChildNodes.getLength(); ++seq_i) {
+							final Node sequenceChildNode = sequenceChildNodes.item(seq_i);
+							if (sequenceChildNode.getNodeName().equals("step")) {
+								final HashMap<String, String> attrSequence = getAttributesMap(sequenceChildNode.getAttributes());
+								final String maxRunCount = attrSequence.get("maxRunCount");
+								final String player = attrSequence.get("player");
+								final Triple<String, String, Integer> newValues = new Triple<String, String, Integer>(attrSequence.get("delegate"), (player==null?"":player), (maxRunCount == null ? 0 : Integer.parseInt(maxRunCount)));
+								s_playerSequence.put(attrSequence.get("name"), newValues);
+							}
+						}
+					}
+				}
+				stepToGo = Math.max(stepToGo, (s_playerSequence.isEmpty() ? 6 : 7));
+			}
+			else if (childNodeName.equals("attatchmentList"))
+			{
+				final NodeList attatchmentListChildNodes = childNode.getChildNodes();
+				for (int p_i = 0; p_i < attatchmentListChildNodes.getLength(); ++p_i)
+				{
+					final Node attatchment = attatchmentListChildNodes.item(p_i);
+					if (attatchment.getNodeName().equals("attatchment"))
+						parseAttatchmentNode(attatchment);
+				}
+				stepToGo = Math.max(Math.max(stepToGo, s_unitAttatchments.isEmpty() ? 8 : 10), s_territoyProductions.isEmpty() ? 0 : 11);

```

Comment:
magic numbers, if anything change there will be a lot to update.  An enum for steps with a static lookup function to get you the step index by name would be of value here.